src/plugins/saved_objects_management/public/register_services.ts

```diff
@@ -80,4 +80,12 @@ export const registerServices = async (
       service: visBuilder.savedVisBuilderLoader,
     });
   }
+
+  if (uiActions) {
```

Comment:
I couldn't find a better way to inject UISettings to the component. The tricky part is that the other services used (notifications, overlays, data, etc.) are coming from core and passed down to the child components. See [here](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/feature/feature-anywhere/src/plugins/saved_objects_management/public/management_section/mount_section.tsx#L62).

But, this component is instantiated in `setup()` and then uses `core.getStartServices()` to get the `start` services of each of those services.

Since UISettings is not part of core, and since we need `UISettingsStart` instead of `UISettingsSetup` (in order to use the `getTrigger()` fn), it couldn't be cleanly passed down here, since we have to call it in `start()` instead of `setup()`. If we add it to the start services like I've done in `SavedObjectsManagementPluginStart` then it can be instantiated and accessed there. Ideally UISettings is part of core and we don't have to jump through these hoops.

Would you recommend another way? I guess we could add a whole `services.ts` file with getters/setters like I've seen as the more common pattern, but i was hesitant to do that since I thought it may be confused with the rest of the "registered services" which look to be just different saved object loaders

Reply:
Got it - I don't think I fully understood `getStartServices()` until now. Thanks for the clarification.

Reply:
I've updated to follow your outlined strategy